src/algebra/continued_fractions/convergents_equiv.lean

```diff
@@ -0,0 +1,415 @@
+/-
+Copyright (c) 2020 Kevin Kappelmann. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Kappelmann
+-/
+import algebra.continued_fractions.continuants_recurrence
+import algebra.continued_fractions.terminated_stable
+import tactic.linarith
+/-!
+# Equivalence of Recursive and Direct Computations of `gcf` Convergents
+
+## Summary
+
+We show the equivalence of two computations of convergents (recurrence relation (`convergents`) vs.
+direct evaluation (`convergents'`)) for `gcf`s on linear ordered fields. We follow the proof from
+[hardy2008introduction], Chapter 10. Here's a sketch:
+
+Let `c` be a continued fraction `[h; (a₀, b₀), (a₁, b₁), (a₂, b₂),...]`, visually:
+                                a₀
+                h + --------------------------
+                                  a₁
+                      b₀ + --------------------
+                                    a₂
+                            b₁ + --------------
+                                        a₃
+                                  b₂ + --------
+                                      b₃ + ...
+
+One can compute the convergents of `c` in two ways:
+1. Directly evaluating the fraction described by `c` up to a given `n`
+2. Using the recurrence
```

Comment:
```suggestion
1. Directly evaluating the fraction described by `c` up to a given `n` (`convergents c n`)
2. Using the recurrence (`convergents' c n`)
```
A little redundant, but it could help.

Reply:
I added it (it's the other way round though, so I added it manually)